Source/WebKit/UIProcess/API/glib/WebKitWebExtensionContext.cpp

```
@@ -0,0 +1,2301 @@
+     */
+    signals[PERMISSION_MATCH_PATTERNS_WERE_DENIED] =
+        g_signal_new("permission-match-patterns-were-denied",
+            G_TYPE_FROM_CLASS(objectClass),
+            G_SIGNAL_RUN_LAST,
+            0, 0, 0,
+            g_cclosure_marshal_VOID__VOID,
+            G_TYPE_NONE, 0);
+
+    /**
+     * WebKitWebExtensionContext::permission-match-patterns-were-granted:
+     * @context: the [class@WebExtensionContext]
+     *
+     * This signal is emitted whenever permission match patterns were granted.
+     *
+     * Since: 2.52
+     */
+    signals[PERMISSION_MATCH_PATTERNS_WERE_GRANTED] =
+        g_signal_new("permission-match-patterns-were-granted",
+            G_TYPE_FROM_CLASS(objectClass),
+            G_SIGNAL_RUN_LAST,
+            0, 0, 0,
+            g_cclosure_marshal_VOID__VOID,
+            G_TYPE_NONE, 0);
+}
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+static void extensionDestroyedCallback(gpointer userData, GObject*)
+{
+    WebKitWebExtensionContext* context = WEBKIT_WEB_EXTENSION_CONTEXT(userData);
+    context->priv->extension = nullptr;
+}
+
+/**
+ * webkit_web_extension_context_new_for_extension:
+ * @extension: a #WebKitWebExtension
+ * @error: return location for error or %NULL to ignore
+ *
+ * Create a new Context for the provided [class@WebExtension].
+ * 
+ * Returns: the newly created context
+ * 
+ * Since: 2.52
+ */
+WebKitWebExtensionContext* webkit_web_extension_context_new_for_extension(WebKitWebExtension* extension, GError** error)
+{
+    auto webExtension = webkitWebExtensionToImpl(extension);
+
+    WebKitWebExtensionContext* object = WEBKIT_WEB_EXTENSION_CONTEXT(g_object_new(WEBKIT_TYPE_WEB_EXTENSION_CONTEXT, nullptr));
+    Ref context = WebKit::WebExtensionContext::create(*webExtension, adoptGRef(object));
+
+    // We only want to return errors that are in the Context error domain here. It is assumed that any errors that came up for the WebKitWebExtension would have been handled before adding to a context
```

> It is assumed that any errors that came up for the WebKitWebExtension would have been handled before adding to a context

Maybe add logging if its a mistake for this to happen.

---

Either we can log it, or we can return an error of some kind.

---

I looked at the test cases again, and it's actually expected behaviour for this to work with a bad extension, depending on the error. For example, we can get an InvalidManifestEntry error on WebExtension (which means injected content might be broken), but the rest should still work.